evalml/pipelines/components/transformers/preprocessing/decomposer.py

```diff
@@ -26,10 +32,126 @@ def __init__(self, parameters=None, component_obj=None, random_seed=0, **kwargs)
+                    )
+                elif not self.parameters["time_index"] in X:
+                    time_index_col = self.parameters["time_index"]
+                    raise ValueError(
+                        f"Too many Datetime features provided in data and provided time_index column {time_index_col} not present in data.",
+                    )
+
+        time_index = pd.DatetimeIndex(dt_col, freq=pd.infer_freq(dt_col)).rename(
+            y.index.name,
+        )
+        return y.set_axis(time_index)
+
     @abstractmethod
     def get_trend_dataframe(self, y):
         """Return a list of dataframes, each with 3 columns: trend, seasonality, residual."""
 
     @abstractmethod
     def inverse_transform(self, y):
         """Add the trend + seasonality back to y."""
+
+    def determine_periodicity(self, X, y, method="autocorrelation"):
+        """Function that uses autocorrelative methods to determine the first, signficant period of the seasonal signal.
+
+        Args:
+            X (pandas.DataFrame): The feature data of the time series problem.
+            y (pandas.Series): The target data of a time series problem.
+            method (str): Either "autocorrelation" or "partial-autocorrelation".  The method by which to determine the
+                first period of the seasonal part of the target signal.  "partial-autocorrelation" should currently not
+                be used.  Defaults to "autocorrelation".
+
+        Returns:
+            (list[int]): The integer numbers of entries in time series data over which the seasonal part of the target data
+                repeats.  If the time series data is in days, then this is the number of days that it takes the target's
+                seasonal signal to repeat. Note: the target data can contain multiple seasonal signals.  This function
+                will only return the first, and thus, shortest period.  E.g. if the target has both weekly and yearly
+                seasonality, the function will only return "7" and not return "365".  If no period is detected, returns [None].
+
+        """
+
+        def _get_rel_max_from_acf(y):
+            acf = sm.tsa.acf(y, nlags=np.maximum(400, len(y)))
+            filter_acf = [acf[i] if (acf[i] > 0) else 0 for i in range(len(acf))]
+            rel_max = argrelextrema(
+                np.array(filter_acf),
+                np.greater,
+                order=5,  # considers 5 points on either side to determine rel max
+            )[0]
+            max_acfs = [acf[i] for i in rel_max]
+            if len(max_acfs) > 0:
+                rel_max = np.array([filter_acf.index(max(max_acfs))])
+            else:
+                rel_max = []
+            return rel_max
+
+        def _get_rel_max_from_pacf(y):
+            pacf = sm.tsa.pacf(y)
+            return argrelextrema(pacf, np.greater)[0]
+
+        def _detrend_on_fly(X, y):
```

Comment:
Sadly, it cannot be run after the component is fit...as the intent is to use the output to fit the component properly.  This does seem a bit like a catch-22 because I think really this is kind of an iterative process.  I am not sure what you mean though for the time index, though.

Reply:
Gotcha. My thinking with the time index is that `set_time_index` can do a lot of extra work if it's run multiple times - ideally there'd be some way we can make sure its called as few times as possible within the component.